src/bin/dual-iir.rs

```diff
@@ -73,8 +75,39 @@ const SAMPLE_TICKS: u32 = 1 << SAMPLE_TICKS_LOG2;
 const SAMPLE_PERIOD: f32 =
     SAMPLE_TICKS as f32 * hardware::design_parameters::TIMER_PERIOD;
 
-#[derive(Clone, Copy, Debug, Tree)]
-pub struct Settings {
+#[derive(Clone, Debug, Tree)]
+pub struct FlashSettings {
+    #[tree(depth(3))]
+    pub dual_iir: RuntimeSettings,
+
+    #[tree(depth(1))]
+    pub net: NetSettings,
+}
+
+impl stabilizer::settings::AppSettings for FlashSettings {
+    fn new(net: NetSettings) -> Self {
+        Self {
+            net,
+            dual_iir: RuntimeSettings::default(),
+        }
+    }
+
+    fn net(&self) -> &NetSettings {
+        &self.net
+    }
+}
+
+impl serial_settings::Settings<4> for FlashSettings {
+    fn reset(&mut self) {
+        *self = Self {
+            dual_iir: RuntimeSettings::default(),
+            net: NetSettings::new(self.net.mac),
+        }
```

Comment:
I intentionally wrote it as replacing self so that we couldn't introduce bugs in the future by adding fields to `self` and then forgetting to update them here.

Reply:
But then `Self { net: NetSettings::new(self.net.mac), ..Default::default() }` is shorter and also covers your worry.

Reply:
We can't do this because `Self` doesn't implement `Default` (and can't) because `NetSettings` doesn't implement it because it needs to contain the MAC.

Reply:
We could soften that and default to a fixed fall-back mac (which we also e.g. do on Thermostat-EEM for some HW versions). But Ok.